tools/check_format_test_helper.py

```diff
@@ -58,6 +61,10 @@ def getInputFile(filename):
 # code.
 def fixFileHelper(filename):
   infile = os.path.join(src, filename)
+  directory = os.path.dirname(filename)
+  if not directory == '' and not os.path.isdir(directory):
+    os.makedirs(directory)
```

Comment:
I'm curious why this is needed. If there's a filename reaching here, shouldn't we expect the directory to exist?

Reply:
At the time I added it because I had issues with the fact that the test file is `api/java_options.proto` and requires `api` to exist, but seems like I can remove this and it still works.

Reply:
The other mkdirs is necessary though to avoid:
```
diff proto_format.proto /source/tools/testdata/check_format/proto_format.proto.gold
Traceback (most recent call last):
  File "./tools/check_format_test_helper.py", line 215, in <module>
    errors += checkAndFixError("api/java_options.proto", "Java proto option")
  File "./tools/check_format_test_helper.py", line 126, in checkAndFixError
    errors = checkFileExpectingError(filename, expected_substring)
  File "./tools/check_format_test_helper.py", line 121, in checkFileExpectingError
    command, status, stdout = runCheckFormat("check", getInputFile(filename))
  File "./tools/check_format_test_helper.py", line 55, in getInputFile
    shutil.copyfile(infile, filename)
  File "/usr/lib/python2.7/shutil.py", line 83, in copyfile
    with open(dst, 'wb') as fdst:
IOError: [Errno 2] No such file or directory: 'api/java_options.proto'
```